cmd/fsi.go

```diff
@@ -108,10 +109,16 @@ func (o *FSIOptions) Unlink() error {
 		printRefSelect(o.ErrOut, o.Refs)
 
 		p := &lib.LinkParams{
-			Dir: o.Refs.Dir(),
 			Ref: ref,
 		}
 
+		dsRef := reporef.DatasetRef{}
+		if err := o.FSIMethods.FSIDatasetForRef(&ref, &dsRef); err != nil {
+			p.Dir = o.Refs.Dir()
```

Comment:
The most common case where `FSIDatasetForRef` returns an error is when the provided reference is unknown (or misspelled). For example, if I run `qri fsi unlink me/not_a_dataset` then we'll get an error and hit this codepath. Because the reference has to be explicitly provided, it's constructed by `NewExplicitRefSelect`, so we're sure that `Dir` is an empty string. That will lead to the same behavior as the original bug: the `fsi.Unlink` function will remove the `.qri-ref` from the current directory. Instead, hitting this error should do nothing aside from displaying the error.